File: Day 1 (1)/Homework/Level 02/Homework2.1.py
# 1) ფუნქცია input() ყოველთვის აბრუნებს ტექსტს (string). 
# როდესაც ვწერთ input() - 0, ეს გამოიწვევს შეცდომას (Error), 
# რადგან ტექსტს ვერ გამოვაკლებთ რიცხვს. 
# თუ გვინდა მათემატიკური მოქმედება, ტექსტი უნდა ვაქციოთ რიცხვად int() ფუნქციით.

# 2) მათემატიკური ოპერაციები:
# +  დამატება (მიმატება)
# -  გამოკლება
# * გამრავლება
# /  გაყოფა (აბრუნებს ათწილადს)
# // მთელი რიცხვის მიღება გაყოფისას (ჭრის ნაშთს)
# %  ნაშთიანი გაყოფა (აბრუნებს მხოლოდ ნაშთს)
# ** ხარისხში აყვანა

# 3) მომხმარებლის მონაცემების შეყვანა
first_name = input("Enter your first name: ")
last_name = input("Enter your last name: ")
age = input("Enter your age: ")
fav_number = input("Enter your favorite number: ")

print("name:", first_name)
print("last name:", last_name)
print("age:", age)
print("favorite number:", fav_number)

print("---") # ვიზუალური დაყოფისთვის

# 4) მათემატიკური ოპერაციები 3 რიცხვზე
num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))
num3 = int(input("Enter third number: "))

print("Sum:", num1 + num2 + num3)
print("Difference:", num1 - num2 - num3)
print("Product:", num1 * num2 * num3)
print("Division:", num1 / num2 / num3)

# 5) განსხვავება Frontend-სა და Backend-ს შორის:
# Frontend (ფრონტენდი) არის ის, რასაც მომხმარებელი ხედავს და რასთანაც ინტერაქციაში შედის (ღილაკები, დიზაინი).
# Backend (ბექენდი) არის "კულისებს მიღმა" მომუშავე ნაწილი - მონაცემთა ბაზები, სერვერები და ლოგიკა.

File: Homework2.1.js
//#3)
console.log("Objective 3");
let inputAge = prompt("Task 3: Enter your age:");
let Age = Number(inputAge);
let AgeList = [];

if (Age >= 18) {
  AgeList.push(true);
} else {
  AgeList.push(false);
}
console.log("Age List:", AgeList);

//#4)
console.log("Objective 4");
let numVal4 = Number(prompt("Task 4: Enter a number (Positive/Negative test):"));
let positive = [];
let negative = [];

if (numVal4 > 0) {
  positive.push(numVal4);
  console.log("Positive Array:", positive);
} else {
  negative.push(numVal4);
  console.log("Negative Array:", negative);
}

//#5)
console.log("Objective 5");
let numVal5 = Number(prompt("Task 5: Enter a number (Even/Odd test):"));
let Even = [];
let Odd = [];

if (numVal5 % 2 === 0) {
  Even.push(numVal5);
  console.log("Even Array:", Even);
} else {
  Odd.push(numVal5);
  console.log("Odd Array:", Odd);
}

//#6)
console.log("Objective 6");
let numbersArray = [10, 20, 30, 40, 50];
console.log("Original array (Task 6):", numbersArray);

numbersArray.shift(); 
console.log("After deleting first element:", numbersArray);


//#7)
console.log("Objective 7");
let colorsArray = ["Red", "Green", "Blue", "Yellow", "Purple"];
console.log("Original array (Task 7):", colorsArray);

colorsArray.pop(); 
console.log("After deleting last element:", colorsArray);

//#8)
console.log("Objective 8");
let Fruit = ["Apple", "Banana", "Orange"];
Fruit.unshift("Mango"); 
console.log("Fruit Array after unshift:", Fruit);

//#9)
console.log("Objective 9");
let num1 = Number(prompt("Calculator: Enter first number:"));
let operator = prompt("Calculator: Enter operator (+, -, *, /):");
let num2 = Number(prompt("Calculator: Enter second number:"));
let result;

if (operator === "+") {
  result = num1 + num2;
} else if (operator === "-") {
  result = num1 - num2;
} else if (operator === "*") {
  result = num1 * num2;
} else if (operator === "/") {
  result = num2 !== 0 ? num1 / num2 : "Error (Cannot divide by zero)";
} else {
  result = "Invalid operator!";
}

console.log(`Calculator Result: ${num1} ${operator} ${num2} = ${result}`);
alert(`Calculator Result: ${num1} ${operator} ${num2} = ${result}`);